Add SSL configuration

USA-RedDragon opened the pull request:
## Description:

Allow the OpenGarage integration to configure the transport layer it uses to communicate (HTTP or HTTPS). See issue #29037 

**Related issue (if applicable):** fixes #29037 

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#11283

## Example entry for `configuration.yaml` (if applicable):
```yaml
# Example configuration.yaml entry
cover:
  platform: opengarage
  covers:
    garage:
      ssl: true
      verify_ssl: false
      host: garage.example.com
      port: 443
      device_key: opendoor
      name: Right Garage Door
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

Commit:
OpenGarage: Add protocol configuration (#29037)

homeassistant commented:
Hi @USA-RedDragon, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#29038).

Once you do that we will be able to review and accept this pull request.

Thanks!

pvizeli commented on homeassistant/components/opengarage/cover.py:
```diff
@@ -42,6 +44,9 @@
         vol.Required(CONF_HOST): cv.string,
         vol.Optional(CONF_NAME): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_PROTOCOL, default=DEFAULT_PROTOCOL): vol.In(
```
Maybe `CONF_SSL`? HTTP and HTTPS are the same protocol but a different transport layer.

USA-RedDragon replied:
Fair point. I'll push a change. Thanks

Commit:
OpenGarage: Change CONF_PROTOCOL to CONF_SSL

Resolves https://github.com/home-assistant/home-assistant/pull/29038#discussion_r350173095

Commit:
OpenGarage: Add `verify_ssl` as an option
